packages/jest/src/transformer.js

```diff
@@ -1,9 +1,17 @@
-const babel = require('babel-core');
+const { transform } = require('@babel/core');
 
 module.exports = {
+  // This is inspired by:
+  // https://github.com/facebook/jest/blob/v22.4.2/packages/babel-jest/src/index.js#L105-L147
+  // And is required due to:
+  // https://github.com/facebook/jest/issues/1468
+  // TODO: See if it would be easier to switch to the higher-level babel-jest,
```

Comment:
You should be able to. If not, please open up an issue 🙂 

If you wanna hold off on that, you should make sure to return a sourcemap here, Jest handles them nowadays (should allow you to drop `retainLines`)

Reply:
Hi @SimenB - thank you for taking a look :-)

Yeah there's lots of potential for clean-up in our current Jest config!

The problem I was finding was that mentioned in facebook/jest#1468, in that the only way to pass the babel options would be to call babel-jest's `createTransformer()` in our own `process()` (since that's the earliest we have access to `config.globals`), which seems a little clunky - plus means we can't use babel-jest's `getCacheKey()` implementation and still have to duplicate `canInstrument: true` etc.

eg:
```js
// transformer.js
const { createTransformer } = require('babel-jest')

module.exports = {
  canInstrument: true,
  process(src, filename, config, transformOptions) {
    return createTransformer(config.globals.BABEL_OPTIONS).process(src, filename, config, transformOptions);
};
```

Though perhaps a better way would be to circumvent `config.globals` and pass our babel config via `process.env` like this?
```js
// transformer.js
const { createTransformer } = require('babel-jest')
const babelConfig = JSON.parse(process.env.BABEL_OPTIONS);

module.exports = createTransformer(babelConfig);
```

Or is there a simpler way I'm missing? (Having to use a custom transformer seems pretty heavyweight for just wanting to not use a `.babelrc`, but I realise we're likely an edge-case :-))

Reply:
I've filed #851 for making these changes.